### ISSUE TITLE
[BUG] npx <gist> leads to command not found

### ISSUE THREAD

**Issue by kentcdodds** [npm/cli#5132]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

I'm trying to run:

```
npx https://gist.github.com/kentcdodds/bb452ffe53a5caa3600197e1d8005733
```

However I get:

```
sh: workshop-setup: command not found
```

For context, I use this for step of my workshop repositories. It checks that learner's machines have the correct versions of things installed, installs dependencies, and runs validation to make sure things were installed correctly. This worked before I updated to npx v8.

### Expected Behavior

I expect the bin of the gist's package.json to be run as previous versions of npx as recent as v7.

### Steps To Reproduce

1. In any environment
2. With any config
3. Run `npx https://gist.github.com/kentcdodds/bb452ffe53a5caa3600197e1d8005733`
4. See the above error

### Environment

- npm: latest
- Node.js: latest
- OS Name: any
- System Model Name: any
- npm config: any

(All my students are running into this. Even my CI: https://github.com/kentcdodds/react-fundamentals/runs/6989184285?check_suite_focus=true)

**Comment by nlf** [comment]
it looks like what's going on here is that the `package.json` in your gist has the name `workshop-setup` but since this package is also published to the npm registry, we're reading the `package.json` file for a name and then installing the package from the registry instead of the gist.

this is very clearly a bug. thanks for the report!

**Comment by kentcdodds** [comment]
That's great! Thank you for looking into this. I tried removing the `name` and setting it to `@kentcdodds/workshop-setup` and neither of those solutions worked. Is there a workaround I can use until this is fixed?

**Comment by gengjiawen** [comment]
> That's great! Thank you for looking into this. I tried removing the `name` and setting it to `@kentcdodds/workshop-setup` and neither of those solutions worked. Is there a workaround I can use until this is fixed?

Another way is downgrade npm. 

```
npm i -g npm@6
```

**Comment by wraithgar** [comment]
Duplicate issue here w/ some more comments: https://github.com/npm/cli/issues/5134

**Comment by wraithgar** [comment]
This should be fixed as of `npm@8.16.0`

```sh
~/D/n/s/empty $ npx gist:a5a65137bd390864e294ec80bb91e005
Need to install the following packages:
  gist:a5a65137bd390864e294ec80bb91e005
Ok to proceed? (y) 
Hi there! I am @gar/npm-exec-gist version 1.0.0
```

**Comment by lopopolo** [comment]
Hey folks, I think a broken version of `npm`/`npx` (18.15.0) has made it into the default GitHub Actions virtual image: 

- https://github.com/actions/runner-images/blob/4de27cdee12be175a0e37116dfd31de733febd46/images/linux/Ubuntu2004-Readme.md#package-management
- https://github.com/actions/runner-images/issues/6136
